paymentapi/src/test/java/fi/hel/verkkokauppa/payment/api/PaytrailPaymentControllerTest.java

```diff
@@ -0,0 +1,200 @@
+
+import fi.hel.verkkokauppa.common.error.CommonApiException;
+import fi.hel.verkkokauppa.payment.api.data.GetPaymentRequestDataDto;
+import fi.hel.verkkokauppa.payment.api.data.OrderDto;
+import fi.hel.verkkokauppa.payment.api.data.OrderItemDto;
+import fi.hel.verkkokauppa.payment.api.data.OrderWrapper;
+import fi.hel.verkkokauppa.payment.model.Payer;
+import fi.hel.verkkokauppa.payment.model.Payment;
+import fi.hel.verkkokauppa.payment.model.PaymentItem;
+import fi.hel.verkkokauppa.payment.repository.PayerRepository;
+import fi.hel.verkkokauppa.payment.repository.PaymentItemRepository;
+import fi.hel.verkkokauppa.payment.repository.PaymentRepository;
+import fi.hel.verkkokauppa.payment.testing.annotations.RunIfProfile;
+import static org.junit.Assert.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@Slf4j
+public class PaytrailPaymentControllerTest {
+
+    @Autowired
+    private PaytrailPaymentController paytrailPaymentController;
+
+    @Autowired
+    private PaymentRepository paymentRepository;
+
+    @Autowired
+    private PayerRepository payerRepository;
+
+    @Autowired
+    private PaymentItemRepository paymentItemRepository;
+
+    private ArrayList<String> toBeDeletedPaymentId = new ArrayList<>();
+
+    @After
+    public void tearDown() {
+        try {
+            toBeDeletedPaymentId.forEach(paymentId -> {
+                log.info(paymentId);
+                paymentRepository.deleteById(paymentId);
+                payerRepository.deleteByPaymentId(paymentId);
+                paymentItemRepository.deleteByPaymentId(paymentId);
+            });
+            // Clear list because all deleted
+            toBeDeletedPaymentId = new ArrayList<>();
+        } catch (Exception e) {
+            log.info("delete error {}", e.toString());
+            toBeDeletedPaymentId = new ArrayList<>();
+        }
+    }
+
+    @Test
+    @RunIfProfile(profile = "local")
+    public void testCreatePaymentFromOrder() {
+        GetPaymentRequestDataDto paymentRequestDataDto = new GetPaymentRequestDataDto();
+        paymentRequestDataDto.setMerchantId("01fde0e9-82b2-4846-acc0-94291625192b");
```

Comment:
Could you create [same function](https://github.com/City-of-Helsinki/verkkokauppa-core/blob/07be80a2d8914bc31bbd5f915303788cbb13f70c/orderapi/src/test/java/fi/hel/verkkokauppa/order/test/utils/TestUtils.java#L262) to here

Reply:
So we dont need to add hard coded merchantId:s

Reply:
Good point! I might add this method to CommonServiceConfigurationClient, so it can be used elsewhere more easily.